packages/shorebird_cli/lib/src/auth/auth.dart

```diff
@@ -114,21 +161,47 @@ class Auth {
   final String _credentialsDir;
   final ObtainAccessCredentials _obtainAccessCredentials;
   final CodePushClientBuilder _buildCodePushClient;
+  String? _token;
 
   String get credentialsFilePath {
     return p.join(_credentialsDir, 'credentials.json');
   }
 
   http.Client get client {
-    final credentials = _credentials;
-    if (credentials == null) return _httpClient;
-    return AuthenticatedClient(
-      credentials: credentials,
+    if (_credentials == null && _token == null) return _httpClient;
+
+    if (_token != null) {
+      return AuthenticatedClient.token(token: _token!, httpClient: _httpClient);
+    }
+
+    return AuthenticatedClient.credentials(
+      credentials: _credentials!,
       httpClient: _httpClient,
       onRefreshCredentials: _flushCredentials,
     );
   }
 
+  Future<AccessCredentials> loginCI(void Function(String) prompt) async {
+    final client = http.Client();
+    try {
+      final credentials = await _obtainAccessCredentials(
+        _clientId,
+        _scopes,
+        client,
+        prompt,
```

Comment:
Doesn't matter, but I think this trailing comma caused this to expand to 5 lines instead of 2 🤷

Reply:
Without the trailing comma the indentation looks a bit off 🤷